src/promptflow/tests/sdk_cli_test/tool_record.py

```diff
@@ -0,0 +1,141 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import base64
+import collections
+import hashlib
+import json
+import os
+from pathlib import Path
+from typing import Dict, OrderedDict
+
+from promptflow._internal import ToolProvider, tool
+from promptflow._utils.dataclass_serializer import serialize
+from promptflow._utils.tool_utils import get_inputs_for_prompt_template
+from promptflow.contracts.run_info import RunInfo as NodeRunInfo
+
+
+class RecordStorage:
+    """
+    RecordStorage static class to manage recording file storage_record.json
```

Comment:
may have more detailed docstring how how this is designed to work.

Reply:
yes, offline syncd